### DISCUSSION TITLE
FR: Proportional Adjustment of Brightness per Bulb in async_turn_on

### DISCUSSION THREAD

**Discussion by Julien-Decoen** [jseidl/magic-areas#428] [Ideas / Feature Requests]
Hello there,

In the async_turn_on method of MA, I’d like to propose an enhancement: when adjusting the room's brightness, rather than setting each bulb to the same brightness level, we could calculate the difference between the current room brightness and the desired level. This difference would then be applied proportionally across each bulb, allowing for a more balanced and natural lighting adjustment.

I’ve already implemented this on my fork, so I’d be happy to handle the coding if we decide to move forward with this approach.

Looking forward to your thoughts!
Lou_Juicy

**Comment by Julien-Decoen** [comment]
I’ve implemented this feature in [PR #429](https://github.com/jseidl/hass-magic_areas/pull/429), along with some simplifications to the devcontainer. Let me know if it works well on your end or if adjustments are needed. Happy to hear your thoughts!

**Comment by jseidl** [comment]
Hey @Julien-Decoen, I just got to know about this integration https://github.com/fredck/lightener from my buddy @tbrasser, looks like it does the same but allows finer tuning, maybe better use that instead?

I'm not against having this feature, BTW, just figured I'd run this integration through you as it might be interesting
